@LinProp/LinProp.m

```diff
@@ -431,161 +431,347 @@ function display(obj)
-                            C = A;
+        %SUBSASGN Subscripted assignment.
+        %   A(I) = B assigns the values of B into the elements of A specified by
+        %   the subscript vector I.  B must have the same number of elements as I
+        %   or be a scalar. 
+        %
+        %   A(I,J) = B assigns the values of B into the elements of the rectangular
+        %   submatrix of A specified by the subscript vectors I and J.  A colon used as
+        %   a subscript, as in A(I,:) = B, indicates all columns of those rows
+        %   indicated by vector I. Similarly, A(:,J) = B means all rows of columns J.
+        %
+        %   A(I,J,K,...) = B assigns the values of B to the submatrix of A specified
+        %   by the subscript vectors I, J, K, etc. A colon used as a subscript, as in
+        %   A(I,:,K) = B, indicates the entire dimension. 
+        %
+        %   For both A(I,J) = B and the more general multi-dimensional 
+        %   A(I,J,K,...) = B, B must be LENGTH(I)-by-LENGTH(J)-by-LENGTH(K)-... , or
+        %   be shiftable to that size by adding or removing singleton dimensions, or
+        %   contain a scalar, in which case its value is replicated to form a matrix
+        %   of that size.
+        
+            if strcmp('.', {S.type})
+                error('Dot indexing is not supported for variables of this type.');   %TODO: This was allowed in  V2.4.7 of the uncLib MATLAB wrapper - but did not serve any function, as far as Dion can see.
+            elseif strcmp('{}', {S.type})
+                error('Brace indexing is not supported for variables of this type.');
+            elseif length(S) > 1
+                error('Invalid array indexing.');    % This type of error should never appear.
+            end
+            
+            % I describes the index-region of A that values are assigned
+            % to. I might be larger than A. In that case, A is extended.
+            I = S.subs;
+            dimI = numel(I);
+            
+            % Convert logical indexes to subscripts
+            isLogicalIndex = cellfun(@islogical, I);
+            I(isLogicalIndex) = cellfun(@find, I(isLogicalIndex), 'UniformOutput', false);
+            
+            % check if non-logical indexes have positive integer values (rounding has no effect and not inf, nan also fails this test).
+            if any(cellfun(@(v) any(ceil(v)~=v | isinf(v) | v <= 0), I(~isLogicalIndex)))
+                error('Array indices must be positive integers or logical values.');
+            end
+            
+            % Special case of null assignment to remove elements
+            if isempty(B)
+                if sum(~strcmp(I, ':')) > 1
+                    error('A null assignment can have only one non-colon index.');
+                else
+                    if dimI == 1
+                        if strcmp(I, ':')
+                            C = []; % TODO: Do we need to destroy/free something?
```

Comment:
On the .NET side we don't need to destroy or free something. This is done by the garbage collector.